### DISCUSSION TITLE
I suggest that the BlackHole installer install two virtual devices.

### DISCUSSION THREAD

**Discussion by akihiro-moriyama** [ExistentialAudio/BlackHole#64] [Feature Requests]
Hello. 

I suggest that the BlackHole installer install two virtual devices.

Reason 1: The SoundFlower installer installs two virtual devices (SoundFlower 2ch, 64ch). SoundFlower users are accustomed to having two virtual devices. If BlackHole has two virtual devices, users can easily migrate to the new environment.

Reason 2: The AudioMIDI app is difficult to use as a sound mixer. LadioCast is an excellent sound mixer. However, in LadioCast, howling occurs when a BlackHole virtual device is specified as both an input device and an output device.

Reason 3: Most users do not have Xcode.

**Comment by devinroth** [comment]
Not familiar with LadioCast. Is it just feedback because the input is being routed to the output?

**Comment by akihiro-moriyama** [comment]
> Is it just feedback

Yes.
I want to use LadioCast for audiomixing, recording, and control volume of BlackHole.

**LadioCast**
[https://apps.apple.com/us/app/ladiocast/id411213048](https://apps.apple.com/us/app/ladiocast/id411213048)
![Screenshot 2020-03-02 20 15 38](https://user-images.githubusercontent.com/19588967/75671773-b2b77600-5cc2-11ea-85fb-97f82875776e.png)

**Comment by devinroth** [comment]
Well of course there will be feedback if it passing the input to the output. That’s is what BlackHole is designed to do.  I’m not sure what the issue is.

**Comment by akihiro-moriyama** [comment]
Some people prefer "CamTwist" (Video mixier app) and "LadioCast" (Audio mixer app) for live streaming. I am also one of them.

Two virtual devices are required for mixing the music of  iTunes and the sound of the built-in microphone. The first virtual device is used to virtualize the iTunes output and connect it to LadioCast. And the second virtual device is used to virtualize the output of LadioCast and connect it to the live streaming app.

Without two Blackhole virtual devices, it is expected that LadioCast users will not be able to migrate to MacOS Catalina. LadioCast users are using two SoundFlower virtual devices and SoundFlower does not run on Catalina.

Distributing two virtual devices is more efficient than proving that one virtual device is sufficient.

**Comment by BLZB0B** [comment]
> Without two Blackhole virtual devices, it is expected that LadioCast users will not be able to migrate to MacOS Catalina. LadioCast users are using two SoundFlower virtual devices and SoundFlower does not run on Catalina.

Soundflower now works on Catalina (I have it and Blackhole installed whilst trying to decide which to keep!)

**Comment by devinroth** [comment]
@akihiro-moriyama Ah that makes much more sense. Thanks.

**Comment by devinroth** [comment]
@BLZB0B Soundflower is based on a deprecated API that will at some point be removed from macOS. Not only that but BlackHole has significantly less CPU usage, less latency, and is less buggy than Soundflower.

**Comment by BLZB0B** [comment]
@devinroth - yeah - I'm not saying Soundflower is better - just pointing out that LadioCast users can migrate to Catalina if they really wanted to - SF isn't the restriction.  Sounds like an exaggerated requirement "BlackHole needs/has to do this because we can't use SF".

Depends if this is something on your roadmap or not I suppose.

**Comment by devinroth** [comment]
@BLZB0B I totally understand now. It unfortunately come down to time. I would definitely be willing to spend more time developing this if i had more funding but for right now I’m doing what i can in my free time. This is a good feature request but it isn’t on the top of my priority list. I’m considering releasing a “pro” version.

**Comment by akihiro-moriyama** [comment]
@BLZB0B Sound good. Thank you for the information.
@devinroth I agree. I think that BH is better than SF.

**Comment by cruller0704** [comment]
I also think that Blackhole 2ch should be installed in addition to 16ch.
This is because 8ch sound is unintentionally recorded in screen recording with internal sounds using Blackhole.
It is a waste of bits in audio encoding.

**Comment by devinroth** [comment]
I'm thinking a good solution to this is have two installers. One for the 16ch and one for a 2ch. What do you think?

**Comment by cruller0704** [comment]
@devinroth It's fine.

**Comment by krono-i2** [comment]
+1
It could be useful to select how many interfaces run and the amount of channels for each one. Similar Rogue Amoeba Loopback.

**Comment by yuriteixeira** [comment]
Any news on this feature? Having multiple black-hole audio interfaces is actually very useful. Right now, I'm using [VB-Cable](https://vb-audio.com/Cable/index.htm) as my secondary virtual audio interface, but I would prefer to use BlackHole, since it's open-source.

**Comment by devinroth** [comment]
I don’t think i’m going to do this. It’s easy enough to run separate installers for the different channel versions.

**Reply by yuriteixeira** [reply]
Got it, but since it's open-source, there is always space to make change, and I took the opportunity to take a stab on it: https://github.com/yuriteixeira/BlackHoleWithAux

Ping @akihiro-moriyama, in case you didn't find a better solution yet :)

**Comment by devinroth** [comment]
I’m confused why this is even an important issue for you. What real benefit does it have?

**Reply by yuriteixeira** [reply]
See https://github.com/ExistentialAudio/BlackHole/discussions/64#discussioncomment-168451
